app/assets/javascripts/user.js

```diff
@@ -223,4 +223,62 @@ $(document).ready(function () {
   $("#read_tou").on("click", function () {
     $("#continue").prop("disabled", !($(this).prop("checked") && $("#read_ct").prop("checked")));
   });
+
+
+  const isTaken /* Record<string, string> */ = {}
+
+  function indicateInvalidDisplayName(isValid /* boolean | undefined*/) {
+    // isValid might also be 'undefined', hence we check explicitly for 'true' and 'false'
+    $("#user_display_name").toggleClass("is-valid", isValid === true)
+    $("#user_display_name").toggleClass("is-invalid", isValid === false)
+  }
+
+  function checkDisplayName(){
+    const displayName = $("#user_display_name").val()
+    if(displayName === ""){
+      indicateInvalidDisplayName(false)
+      return;
+    }
+    if(displayName.match(/[\/;.,?%#]/)!== null) {
+      // We detected an invalid character
+      indicateInvalidDisplayName(false)
+      return
+    }
+    indicateInvalidDisplayName(!isTaken[displayName])
+    if(isTaken[displayName] !== undefined){
+      return
+    }
+
+    // We fetch the userpage, as it is the only endpoint where we can use the display name
+    // All (proper) API-calls use the user ID number
+    // To lighten the load, we use the HEAD method
+    fetch("./" + encodeURIComponent(displayName), {method: "HEAD"}).then(
+      response => {
+        if (response.status === 404) {
+          isTaken[displayName] = false
+        }
+        if (response.status === 200) {
+          isTaken[displayName] = true
+        }
+
+        if(isTaken[displayName] === undefined){
+          return
+        }
+        // The input field value might have changed by now, so we check if the current value is still the value we did the request for
+        if(displayName === $("#user_display_name").val()){
+          indicateInvalidDisplayName(!isTaken[displayName])
+        }
+      },
+    ).catch(reason => console.warn("Could not fetch info about", displayName))
+  }
+
+  $("#user_display_name").on("change", function () {
+    checkDisplayName()
+  })
+
+  $("#user_display_name").on("input", function () {
+    checkDisplayName()
+  })
```

Comment:
Isn't only `input` handler enough?

Reply:
In practice, yes. However, a piece of code (e.g. a password manager) might insert a value programmatically, which might not trigger 'input'

Reply:
Did you found such a software?
There are keyboards (iPhone? IPhone landscape mode?) I saw which manipulates the string and pushes the value in one go. These did not triggered "change". But I did not found an interaction which does not trigger "input".